cfme/automate/import_export.py

```diff
@@ -1,70 +1,112 @@
-                .first()
-            return cls(url=url, verify_ssl=verify_ssl > 0, appliance=appliance)
-        except ValueError:
-            raise ValueError('No such repository in the database')
+        url = Input(name="git_url")
+        username = Input(name="git_username")
+        password = Input(name="git_password")
+        verify_ssl = Checkbox(name="git_verify_ssl")
+        submit = Button(id="git-url-import")
+
+    export_all = Button(title="Export all classes and instances")
+    reset_all = Button(title="Reset all components in the following domains: RedHat, ManageIQ")
+    reset_title = Text(
+        ".//div[contains(@class, 'import-or-export')]/h3[contains(text(), 'Reset all')]"
+    )
 
     @property
-    def fill_values_repo_add(self):
-        return {
-            k: v
-            for k, v
-            in {
-                'url': self.url,
-                'username': self.username,
-                'password': self.password,
-                'verify_ssl': self.verify_ssl}.items() if v is not None}
-
-    def fill_values_branch_select(self, branch, tag):
-        """Processes the args into a dictionary to be filled in the selection dialog."""
-        if branch and tag:
-            raise ValueError('You cannot pass branch and tag together')
-        elif tag is not None:
-            return {'type': 'Tag', 'tag': tag}
-        else:
-            return {'type': 'Branch', 'branch': branch}
+    def is_displayed(self):
+        title = "Reset all components in the following domains: RedHat, ManageIQ"
+
+        return (
+            self.in_import_export
+            and self.export_all.is_displayed
+            and self.reset_title.text == title
+        )
+
+
+class FileImportSelectorView(AutomateImportExportBaseView):
+    import_into = BootstrapSelect(id="selected_domain_to_import_to")
+    import_from = BootstrapSelect(
+        locator=".//div[contains(@class, 'bootstrap-select importing-domains')]"
+    )
+    toggle_all = Checkbox(id="toggle-all")
+
+    commit = Button("Commit")
+    back = Button("Back")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_import_export
+            and self.import_from.is_displayed
+            and self.import_into.is_displayed
+        )
+
+
+class GitImportSelectorView(AutomateImportExportBaseView):
+    branch_tag = BootstrapSelect("branch_or_tag")
+    branch = BootstrapSelect(locator='.//div[contains(@class, "bootstrap-select git-branches")]')
+    tag = BootstrapSelect(locator='.//div[contains(@class, "bootstrap-select git-tags")]')
+
+    submit = Button("Submit")
+    cancel = Button("Cancel")
+
+    @property
+    def is_displayed(self):
+        return self.in_import_export and self.branch_tag.is_displayed
+
+
+@attr.s
+class AutomateGitRepository(BaseEntity):
+    url = attr.ib()
+    username = attr.ib(default=None)
+    password = attr.ib(default=None)
+    verify_ssl = attr.ib(default=None)
```

Comment:
No specific reason. but when we take some entity to fill and default is `None` it will not force to fill means it will ignore widget but if we put something default then `fill` method will force to fill widget.

If you want with specific reason; i will change it.

Reply:
@digitronik I was thinking from the ssl_validation point of view. If I use https url and I set ssl to false, it wouldn't cause harm. If I pass http url and set this to true that will fail. So setting false seems to be a logical choice for default. 
But not gonna force you to change. If None works for both kind of urls, then leave it as is.

Reply:
@digitronik I was thinking from the ssl_validation point of view. If I use https url and I set ssl to false, it wouldn't cause harm. If I pass http url and set this to true that will fail. So setting false seems to be a logical choice for default. 
But not gonna force you to change. If None works for both kind of urls, then leave it as is.